backend-api/src/functions/asyncActiveSession/tokenService/tests/tokenService.test.ts

```diff
@@ -122,6 +125,167 @@ describe("Token Service", () => {
             });
           });
         });
+
+        describe("Given the response is not in the shape of a public key", () => {
+          it("Returns an error response", async () => {
+            mockFetch = jest.spyOn(global, "fetch").mockImplementation(() =>
+              Promise.resolve({
+                status: 200,
+                ok: true,
+                json: () =>
+                  Promise.resolve({
+                    keys: ["mockNotAnObject"],
+                  }),
+              } as Response),
+            );
+
+            const result = await tokenService.getSubFromToken(
+              "https://mockJwksEndpoint.com",
+              "mockJwe",
+              100,
```

Comment:
Should I make this even smaller for the tests? Could do 10 or even 1ms delay to speed up unit tests?